### DISCUSSION TITLE
Reporting capabilities

### DISCUSSION THREAD

**Discussion by arung2207** [serenity-is/Serenity#6402] [Help] [answered]
I saw a demo where we can export the information shown on a grid into a pdf but is it possible to create a report template where we can display data at runtime. On the lines of a reporting tool such as Crystal Reports? Is there a way we can import the CR .rpt file and pass data to it and create a pdf?

**Comment by edwardch** [comment, marked as answer]
Hey @arung2207, I created a wiki a while back on [Integrating Telerik Reporting with Serenity](https://github.com/serenity-is/Serenity/wiki/Reporting-Integrating-Telerik-Reporting).

I imagine that it will be very similar to Crystal Reports implementation.

What I have also set up, which isn't in the wiki, is that a user can upload a report template into the uploads directory and the report becomes available to app users immediately to interact with and download in various formats. I'm sure a lot more is possible, though...

**Reply by arung2207** [reply]
Sure thanks for the pointer, will evaluate.
